### ISSUE TITLE
[Display Problem]

### ISSUE THREAD

**Issue by dordio** [fdw/rofimoji#131]
* What does rofimoji show instead of the emojis? Can you please share a screenshot (`Alt+Shift+s** in rofimoji)
Here is a screenshot of rofimoji at work. Sorry for it being a photo, my screenshot software is having one of those days. No icons are displayed
![20220914_1708542](https://user-images.githubusercontent.com/3708447/190209686-e06df60a-4427-44ed-bcfd-9e89c5cccb8c.jpg)

* What do other apps like Firefox/Chrome/your text editor show instead of emojis?
All other apps show the emojis correctly, even when I copy/paste them from rofimoji. I can call rofimoji type a description of what I wan't, copy and paste it every where with no issue.

* Which emoji fonts do you have installed?
I have Noto Emoji installed.

* Are the emoji fonts set in your font config?
I'm not sure about this, but the fonts do appear in font viewer.

* Which versions of rofimoji and pango are installed?
rofimoji is version 5.5.0
as for pango when run "apt list --installed" I get the following list:
gir1.2-pango-1.0/jammy,now 1.50.6+ds-2 amd64 [instalado,automático]
libcogl-pango20/jammy,now 1.22.8-3build1 amd64 [instalado,automático]
libpango-1.0-0/jammy,now 1.50.6+ds-2 amd64 [instalado,automático]
libpangocairo-1.0-0/jammy,now 1.50.6+ds-2 amd64 [instalado,automático]
libpangoft2-1.0-0/jammy,now 1.50.6+ds-2 amd64 [instalado,automático]
libpangomm-1.4-1v5/jammy,now 2.46.2-1 amd64 [instalado,automático]
libpangoxft-1.0-0/jammy,now 1.50.6+ds-2 amd64 [instalado,automático]

Thank you for your time and for your work.

Best regards.

**Comment by fdw** [comment]
This looks like a font problem with `rofi`. Do you set any font in its config?

**Comment by dordio** [comment]
The only font config I have for rofi is this line in i3 config file
`bindsym $mod+d exec --no-startup-id rofi -show run -width 100 -font "Fira Code 18"`

I tried removing the font call, but to no avail, still no icons.

**Comment by fdw** [comment]
Do you use `rofimoji` standalone or as a `rofi` mode?

Apart from that, it looks like a font/display issue, as the correct character is selected. But currently, I have absolutely no idea what could be causing this. If you pass some emoji to `rofi -dmenu`, do they show up?

**Comment by dordio** [comment]
I'm using `rofimoji` standalone, I call it with `bindsym $mod+period exec --no-startup-id rofimoji`

I don't know how to pass emojis to rofi. Sorry.

I'm going to try and reinstall all my fonts, see if it helps.

**Comment by fdw** [comment]
> I don't know how to pass emojis to rofi. Sorry.

This should do the trick: `echo "🙂" | rofi -dmenu`

**Comment by dordio** [comment]
Ok, I'm stupid, I realize that as soon as I read the word "echo". Yeah, I know how to pass something to something else. Don't know what my brain was doing at the time. Sorry.

Anyway I tried passing an emoji to rofi, all i got is an empty rofi window.

I did reinstall my fonts but to no avail.

rofi is version 1.7.1
python is version 3.10.4

**Comment by fdw** [comment]
So `rofimoji` isn't the problem, but something else (I don't think that helps you 🙈). I've found something [here](https://github.com/davatorium/rofi/issues/1042), but that doesn't have a solution, either. Apparently it's related to pango, though.

**Comment by dordio** [comment]
So the site you referred made test the following: adding the to rofi's config file got me the following result (again sorry for the photo, my screenshot app simply won't capture rofimoji)

![20220916_232413](https://user-images.githubusercontent.com/3708447/190824585-ad86d8c9-6dd3-4c0d-b917-a35bd4fe3b36.jpg)

The font I added is `Emoji One`

I'll try to see if I can find any problems related to pango.

**Comment by fdw** [comment]
You can make screenshots of `rofi` with `Alt`+`Shift`+`s` 🙂

Also, that very much looks like a font issue. Maybe [this](https://github.com/stove-panini/fontconfig-emoji/) also helps?

**Comment by dordio** [comment]
Nope, `Alt+Shift+s` doesn't work. I'm starting to think it's an OS issue. I'm running i3 on top of POP OS. I'm considering reinstalling to see if it gets fixed. Since I need the computer for work I might do it next weekend.

Either way the page you refered also didn't help (sad face emoji here).

So, for now, I'm giving up on the issue. If you don't mind I'll comment once I manage to find a solution.

Thank you for all your trouble.

**Comment by fdw** [comment]
Good luck then! 🤞

**Comment by dordio** [comment]
Thank you! (smiley face emoji)(thumbs up emoji)

**Comment by fdw** [comment]
I'll close this for now, but feel free to re-open or open a new issue if you still have trouble 🙂

**Comment by ClashTheBunny** [comment]
I had this same issue (including the linked rofi-only aspect - https://github.com/davatorium/rofi/issues/1042, with just an echo of an emoji into rofi) and it was a problem with Google's NotoColorEmoji.ttf downloaded from both google/fonts and googlefonts/noto-emoji.  I removed that (which I had installed separately) and now emoji are back in rofimoji.

Here are the Noto Emoji now installed on my host:
```
$ fc-list : family file | grep -i noto | grep -i emoji
/usr/local/share/fonts/chromeos/notofonts-20161129/NotoColorEmoji.ttf: Noto Color Emoji
/usr/local/share/fonts/chromeos/notofonts-20161129/NotoEmoji-Regular.ttf: Noto Emoji
/usr/share/fonts/truetype/noto/NotoColorEmoji.ttf: Noto Color Emoji
```

I found this by strace'ing rofi and just randomly removing fonts that showed up until it worked again.
